### PR TITLE
EDSC-4171: Add a new metric/event for granule filters

### PR DESCRIPTION
**Pull request by eudoroolivares2016**
# Overview

### What is the feature?

Adds metrics for the granule filters to be tracked by Google analytics. When the different granule filters are selected a new event is fired to the data layer which can be picked up by Goggle analytics/Google Tag Manager and then used for data analysis.

Some components and tests were updated for the purpose of technical debt reduction

### What is the Solution?

Adding events that fire when each granule filter is selected.

Converted the `GranuleFilterContainer` into a functional component and updated its test into RTL

Spinoff ticket intended for the tests in `GrauleFilterForm` as a standalone due to the size

### What areas of the application does this impact?

The `GranuleFilterForm`

# Testing

### Reproduction steps

- **Environment for testing:*prod*
- **Collection to test with:*Any with granules but, specific ones for certain form items to appear see steps below *

1. Ensure thorough regression testing for the GranuleFiltersForm to ensure no additional unintended functionality was introduced with the metric additions or with the conversion to functional component

2. Ensure that clearing the filters, selecting specific filters, and filtering specific granules (As well as undoing the filtered granules) work as expected for step 1

3. Ensure that on Google tag manager in the preview mode we can see events firing with the value that they have on each of the granule filters available the following is a map of collections which have metadata needed to drive the form

C1595422627-ASF - has orbit information
C2484086411-LPCLOUD - has a tile grid system and equatorial crossing selection
C2617126679-POCLOUD -has cloud cover
C2207478611-FEDEO - is an openSearch collection so only the temporal filter should render on the form for this

4. Since we are updating `formik` to the latest via 2 minor release bumps as part of this tech debt reduction we should regression test the other form using `formik` I did check the diff of the versions on their release notes nothing looked like it would break our implementation but, still.

The other forms using `formik` are
`AdvancedSearchForm` and `RegionSearch` 
For RegionSearch I am just using the `HUC-ID` in Advanced Search

### Attachments

Please include relevant screenshots or files that would be helpful in reviewing and verifying this change.

# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: static/src/js/components/TemporalSelection/TemporalSelection.jsx [hunk 1]
```diff
@@ -227,7 +227,7 @@ export class TemporalSelection extends Component {
               controlId={`${controlId}__recurring`}
             >
               <Form.Check>
-                <Form.Check.Input disabled={enableRecurring} type="checkbox" onChange={onRecurringToggle} checked={isRecurring} />
+                <Form.Check.Input aria-label="is-recurring-checkbox" disabled={enableRecurring} type="checkbox" onChange={onRecurringToggle} checked={isRecurring} />
```
Comment (eudoroolivares2016) [inline static/src/js/components/TemporalSelection/TemporalSelection.jsx:230]: This is a dev artifact while converting to RTL we should either use it in the enzyme test or not

File: static/src/js/containers/GranuleFiltersContainer/__tests__/GranuleFiltersContainer.test.js [hunk 2]
```diff
@@ -117,133 +190,84 @@ describe('mapStateToProps', () => {
...
-          expect(onClearGranuleFiltersMock).toHaveBeenCalledTimes(1)
+          expect(onClearGranuleFilters).toHaveBeenCalledTimes(1)
         })
 
-        test('sets the granuleFiltersNeedsReset flag to false', () => {
-          const { enzymeWrapper, props } = setup()
-
-          enzymeWrapper.setProps({ granuleFiltersNeedsReset: true })
-          enzymeWrapper.update()
+        test('calls onClearGranuleFilters ', () => {
```
Comment (eudoroolivares2016) [inline static/src/js/containers/GranuleFiltersContainer/__tests__/GranuleFiltersContainer.test.js:206]:
```suggestion
        test('calls handleReset ', () => {
```

File: static/src/js/containers/GranuleFiltersContainer/GranuleFiltersContainer.jsx [hunk 3]
```diff
@@ -13,6 +13,7 @@ import { getFocusedCollectionGranuleQuery } from '../../selectors/query'
 import { getFocusedCollectionMetadata } from '../../selectors/collectionMetadata'
 
 import GranuleFiltersForm from '../../components/GranuleFilters/GranuleFiltersForm'
+import { metricsGranuleFilter } from '../../middleware/metrics/actions'
 
 export const mapStateToProps = (state) => ({
```
Comment (eudoroolivares2016) [inline static/src/js/containers/GranuleFiltersContainer/GranuleFiltersContainer.jsx:18]: I don't think that ` temporal: state.query.collection.temporal` is actually being used anymore

**Review by dpesall — APPROVED** [review]

**Review by rushgeo — APPROVED** [review]

**Review by daniel-zamora — APPROVED** [review]

File: static/src/js/components/GranuleFilters/GranuleFiltersForm.jsx [hunk 4]
```diff
@@ -399,6 +440,13 @@ export const GranuleFiltersForm = (props) => {
                       }
                     }
 
+                    // Take metric when the isRecurring toggle is turned on
+                    onMetricsGranuleFilter({
+                      type: 'Set Recurring',
+                      value: isChecked
+                    })
+
+                    // TODO why here and not the other
```
[the remark this answers is not in the document]
Comment (eudoroolivares2016) [reply static/src/js/components/GranuleFilters/GranuleFiltersForm.jsx:449]: When I work the test ticket to update RTL I'll try to drop these, I just have concerns that these timeout wrappers serve some forgotten purpose for certain events

**Review by daniel-zamora — APPROVED** [review]

**Pull request merged**